### PR TITLE
Sam/aave fee swap

### PR DESCRIPTION
**Pull request by samholmes**
#### PR Dependencies

https://github.com/EdgeApp/edge-react-gui/pull/3695

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202886386261596

### CODE REVIEW

File: src/hooks/useAsyncValue.ts [hunk 1]
```diff
@@ -6,20 +6,20 @@ import { useAsyncEffect } from './useAsyncEffect'
  * Returns the value of an async function or an error.
  * Re-runs when its dependencies change, just like `useAsyncEffect`.
  */
-export function useAsyncValue<T>(effect: () => Promise<T>, deps?: unknown[]): [T | null, Error | null] {
-  const [value, setValue] = React.useState<T | null>(null)
-  const [error, setError] = React.useState<Error | null>(null)
+export function useAsyncValue<T>(effect: () => Promise<T>, deps?: unknown[]): [T | undefined, Error | undefined] {
+  const [value, setValue] = React.useState<T | undefined>(undefined)
```
Comment (Jon-edge) [inline src/hooks/useAsyncValue.ts:10]: That's pretty neat, didn't know that!

**Review by Jon-edge — CHANGES_REQUESTED** [review]

File: src/components/scenes/Loans/LoanCreateConfirmationScene.tsx [hunk 2]
```diff
@@ -44,6 +45,7 @@ export const LoanCreateConfirmationScene = (props: Props) => {
 
   const [[actionProgram, networkFeeAmountAggregate = '0'] = [], actionProgramError] = useAsyncValue(async () => {
     const borrowPluginId = borrowPlugin.borrowInfo.borrowPluginId
+    const borrowWalletNativeBalance = borrowEngineWallet.balances[borrowEngineWallet.currencyInfo.currencyCode]
```
Comment (Jon-edge) [inline src/components/scenes/Loans/LoanCreateConfirmationScene.tsx:48]: use useWalletBalance hook?

**Review by Jon-edge — CHANGES_REQUESTED** [review]

**Comment by Jon-edge** [comment]
Looks good, but I noticed that there's no updates to the Loan Status display. Can we simply concatenate the display messages of the parallel actions for now? LoanStatusScene looks weird right now if you test it

**Review by Jon-edge — COMMENTED** [review]
Looks good, but I noticed that there's no updates to the Loan Status display. Can we simply concatenate the display messages of the parallel actions for now? LoanStatusScene looks weird right now if you test it

**Review by Jon-edge — CHANGES_REQUESTED** [review]
Looks good, but I noticed that there's no updates to the Loan Status display. Can we simply concatenate the display messages of the parallel actions for now? LoanStatusScene looks weird right now if you test it

File: src/hooks/useAsyncValue.ts (hunk 1 above)
Comment (samholmes) [reply src/hooks/useAsyncValue.ts:10]: 👍

File: src/components/scenes/Loans/LoanCreateConfirmationScene.tsx (hunk 2 above)
Comment (samholmes) [reply src/components/scenes/Loans/LoanCreateConfirmationScene.tsx:48]: You can't use hooks within callbacks.

**Review by Jon-edge — APPROVED** [review]
Approved after comments as discussed in our call:
-Would be nice to begin hiding away more implementations of things like fee calculations below of the GUI layer
-Update the dependencies for acf1fda30ea3d900675d479650a358fe94b6db0f to watch for wallet balances balances

**Pull request merged**